internal/datastore/spanner/watch.go

```diff
@@ -102,10 +106,22 @@ func (sd spannerDatastore) loadChanges(
 			&tpl.Subject.Namespace,
 			&tpl.Subject.ObjectId,
 			&tpl.Subject.Relation,
+			&caveatName,
+			&caveatCtx,
 		)
 		if err != nil {
 			return err
 		}
+		if caveatName.Valid && caveatCtx.Valid {
+			caveatStruct, err := structpb.NewStruct(caveatCtx.Value.(map[string]interface{}))
```

Comment:
Is the `JSON` type from Spanner a map[string]any under the hood, so we know this type assertion won't panic?

also super nit 😄 
```suggestion
			caveatStruct, err := structpb.NewStruct(caveatCtx.Value.(map[string]any))
```